### PR TITLE
Destructure hover

### PR DESCRIPTION
**Pull request by gabydd**
This implements type resolution for assign destructure using the right hand side and the index, it works for the uses I have tried but if there is a different way zls wants to approach this thats totally fine to me. I am also quite new to zig so this might not be the most idiomatic way to do this.

### CODE REVIEW

File: src/analysis.zig [hunk 1]
```diff
@@ -3660,7 +3660,25 @@ pub const DeclWithHandle = struct {
...
+                }) orelse return null;
+                return switch (node.data) {
+                    .array => blk: {
+                        var buffer: [2]Ast.Node.Index = undefined;
+                        const array = tree.fullArrayInit(&buffer, rhs) orelse return null;
+                        break :blk try analyser.resolveTypeOfNode(.{
+                            .node = array.ast.elements[pay.index],
+                            .handle = self.handle,
+                        });
+                    },
```
Comment (Techatrix) [inline src/analysis.zig:3677]: The child type of the array has already been resolved.
```suggestion
                    .array => |array_info| try array_info.elem_ty.instanceTypeVal(analyser),
```

File: src/analysis.zig [hunk 2]
```diff
@@ -3660,7 +3660,25 @@ pub const DeclWithHandle = struct {
...
+                    .other => try analyser.resolveTupleFieldType(node, pay.index),
+                    else => null,
+                };
+            }, // TODO
```
Comment (Techatrix) [inline src/analysis.zig:3681]: TODO solved.
```suggestion
            },
```

**Review by Techatrix — CHANGES_REQUESTED** [review]

File: src/analysis.zig (hunk 1 above)
Comment (Techatrix) [reply src/analysis.zig]: I should have also mentioned that this makes the `rhs` variable above unused.

Comment (gabydd) [reply src/analysis.zig]: yeah just fixing that manually and testing will push in a second

**Comment by Techatrix** [comment]
While testing this, I realized that adding an explicit type specifier to the variable was not being used to resolve the type.
Here is an example:
```zig
test {
    const bar, const baz: u32 = undefined;
}
```
Hovering over `baz` was not showing `u32`. I have fixed this and pushed the change to your PR.

**Review by Techatrix — APPROVED** [review]
Thank you very much.

**Pull request merged**